rotateRad.go

```diff
@@ -5,8 +5,8 @@ import (
 )
 
 // RotateRad returns the vector obtained by rotating
-// the specified 2D vector by the angle specified in
-// radians.
+// the specified 2D vector anticlockwise by the angle
+// specified in radians.
 func (v V2F[T]) RotateRad(angleRad float64) V2F[float64] {
```

Comment:
It should return same vector type for float vectors:

```go
func (v V2F[T]) RotateRad(angleRad float64) V2F[T] {
```

This might need to fix the type casting in the implementation.

Reply:
Right, will cast the resulting X and Y to type T.